File: src/Python/101-200/119.YangHuiTriangleII.py

class Solution:
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        rowIndex += 1
        if rowIndex == 0:
            return []
        init = [1]
        for i in range(rowIndex - 1):
            init.append(0)
            init.insert(0, 0)
            for j in range(len(init) - 1):
                init[j] = init[j] + init[j + 1]
        # return init[0 : rowIndex]
        return init


if __name__ == '__main__':
    solution = Solution()
    print(solution.getRow(0))
    print(solution.getRow(1))
    print(solution.getRow(2))
    print(solution.getRow(3))
    print(solution.getRow(4))
    print(solution.getRow(5))
else:
    pass

File: src/Java/101-200/119.YangHuiTriangleII.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;

class Solution {
    public List<Integer> getRow(int rowIndex) {
        ++rowIndex;
        ArrayList<Integer> init = new ArrayList<>();
        if (rowIndex == 0)
            return init;
        init.add(1);
        for (int i = 0; i < rowIndex - 1; i++) {
            init.add(0);
            init.add(0, 0);
            for (int j = 0; j < init.size() - 1; j++) {
                init.set(j, init.get(j) + init.get(j + 1));
            }
        }
        init.removeIf((i1) -> {return i1 == 0;});
        return init;
    }
}
